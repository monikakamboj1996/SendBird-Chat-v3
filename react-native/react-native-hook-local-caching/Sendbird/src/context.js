import React, { createContext } from 'react';

export const AppContext = createContext();

export const withAppContext = (Component, mapStateToProp = null) => {
  const ConsumableComponent = props => {
    return (
      <AppContext.Consumer>
        {state => {
          const mappedProps = mapStateToProp ? mapStateToProp(state) : state;
          const mergedProps = { ...props, ...mappedProps };
          return <Component {...mergedProps} />;
        }}
      </AppContext.Consumer>
    );
  };
  return ConsumableComponent;
};
