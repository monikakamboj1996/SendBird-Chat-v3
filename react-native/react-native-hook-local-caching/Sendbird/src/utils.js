import AsyncStorage from '@react-native-async-storage/async-storage';
import { CommonActions } from '@react-navigation/native';

import notifee from '@notifee/react-native';

const channelNameMaxMembers = 3;
const channelNameEllipsisLength = 32;
const maxUnreadMessageCount = 99;

export const ellipsis = (s, len) => {
  return s.length > len ? s.substring(0, len) + '..' : s;
};
export const createChannelName = channel => {
  if (channel.name === 'Group Channel' || channel.name.length === 0) {
    const nicknames = channel.members.map(m => m.nickname);
    if (nicknames.length > channelNameMaxMembers) {
      return ellipsis(
        `${nicknames.slice(0, channelNameMaxMembers + 1).join(', ')} and ${
          nicknames.length - channelNameMaxMembers
        } others`,
        channelNameEllipsisLength,
      );
    } else {
      return ellipsis(`${nicknames.join(', ')}`, channelNameEllipsisLength);
    }
  }
  return ellipsis(channel.name, channelNameEllipsisLength);
};
export const createUnreadMessageCount = channel => {
  if (channel.unreadMessageCount > maxUnreadMessageCount) {
    return `${maxUnreadMessageCount}+`;
  } else {
    return `${channel.unreadMessageCount}`;
  }
};
export const onRemoteMessage = async remoteMessage => {
  // Set the channel for Android
  const channelId = await notifee.createChannel({
    id: 'SendbirdNotificationChannel',
    name: 'Sendbird RN Sample',
  });

  if (remoteMessage && remoteMessage.data) {
    let pushActionId = 'SendbirdNotification-';

    // Set the notification push action id from channel url
    const message = JSON.parse(remoteMessage.data.sendbird);
    let channelUrl = null;
    if (message && message.channel) {
      channelUrl = message.channel.channel_url;
    }
    pushActionId += channelUrl;

    await AsyncStorage.setItem(pushActionId, JSON.stringify(remoteMessage));

    // Display a notification
    await notifee.displayNotification({
      title: 'Sendbird Sample',
      body: remoteMessage.data.message,
      android: {
        channelId,
        pressAction: {
          id: pushActionId,
          launchActivity: 'default',
        },
      },
    });
  }
};

export const handleNotificationAction = async (navigation, sendbird, currentUser) => {
  // Move to channel from pushed notification
  const initialNotification = await notifee.getInitialNotification();
  if (initialNotification && initialNotification.pressAction) {
    const remoteMessage = JSON.parse(await AsyncStorage.getItem(initialNotification.pressAction.id));
    if (remoteMessage && remoteMessage.data) {
      const message = JSON.parse(remoteMessage.data.sendbird);
      if (message && message.channel) {
        const channel = await sendbird.GroupChannel.getChannel(message.channel.channel_url);
        navigation.dispatch(state => {
          const lobbyIndex = state.routes.findIndex(route => route.name === 'Lobby');
          const newRoute = { name: 'Chat', params: { channel, currentUser } };
          const routes = [...state.routes.slice(0, lobbyIndex + 1), newRoute];
          const action = CommonActions.reset({ ...state, routes, index: routes.length - 1 });

          const chatRoute = state.routes.find(route => route.name === 'Chat');
          if (chatRoute && chatRoute.params && chatRoute.params.channel) {
            if (chatRoute.params.channel.url === channel.url) {
              // Navigate from same channel
              return CommonActions.reset(state);
            } else {
              // Navigate from another channel
              return action;
            }
          } else {
            // Navigate from Lobby
            return action;
          }
        });
        await AsyncStorage.removeItem(initialNotification.pressAction.id);
      }
    }
  }
};
