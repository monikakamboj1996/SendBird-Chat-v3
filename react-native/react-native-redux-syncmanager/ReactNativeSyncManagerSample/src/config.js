
export default class Config {
	static get appId() {
		return '9DA1B1F4-0BE6-4DA8-82C5-2E81DAB56F23';
	}
}