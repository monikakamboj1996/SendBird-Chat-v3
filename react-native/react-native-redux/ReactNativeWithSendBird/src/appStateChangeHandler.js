export default (function() {
  var instance;

  function AppStateChangeHandler() {
    this.cbs = {};
    this.addCallback = (key, cb) => {
      this.cbs[key] = cb;
      return () => {
        delete this.cbs[key];
      };
    };

    this.notify = () => {
      for (let key in this.cbs) {
        this.cbs[key]();
      }
    };
  }

  return {
    getInstance: function() {
      if (!instance) {
        instance = new AppStateChangeHandler();
      }
      return instance;
    }
  };
})();
