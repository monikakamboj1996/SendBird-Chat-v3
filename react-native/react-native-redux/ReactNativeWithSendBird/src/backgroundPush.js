import firebase from 'react-native-firebase';

export default async message => {
  try {
    const text = message.data.message;
    const payload = JSON.parse(message.data.sendbird);
    const localNotification = new firebase.notifications.Notification({
      show_in_foreground: true
    }).android
      .setChannelId('com.reactnativewithsendbird.default_channel_id')
      .android.setSmallIcon('sendbird_ic_notification')
      .android.setPriority(firebase.notifications.Android.Priority.High)
      .setNotificationId(message.messageId)
      .setTitle('New message')
      .setSubtitle(`Unread message: ${payload.unread_message_count}`)
      .setBody(text)
      .setData(payload);
    return firebase.notifications().displayNotification(localNotification);
  } catch (e) {
    return Promise.resolve();
  }
};
