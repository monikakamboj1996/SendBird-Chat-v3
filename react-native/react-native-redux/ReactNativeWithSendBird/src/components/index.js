export * from './Spinner';
export * from './Message';
export * from './MessageItem';
export * from './ImageItem';
export * from './MessageInput';
export * from './Hr';
export * from './MessageContainer';
export * from './MessageAvatar';
export * from './MessageBubble';
export { Button, Icon, Avatar, FormLabel, FormInput, FormValidationMessage, ListItem } from 'react-native-elements';
