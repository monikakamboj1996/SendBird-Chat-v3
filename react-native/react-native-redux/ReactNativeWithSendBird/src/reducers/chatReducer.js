import {
  INIT_CHAT_SCREEN,
  CREATE_CHAT_HANDLER_SUCCESS,
  CREATE_CHAT_HANDLER_FAIL,
  CHANNEL_TITLE_CHANGED,
  CHANNEL_TITLE_CHANGED_FAIL,
  MESSAGE_LIST_SUCCESS,
  MESSAGE_LIST_FAIL,
  SEND_MESSAGE_TEMPORARY,
  SEND_MESSAGE_SUCCESS,
  SEND_MESSAGE_FAIL,
  CHANNEL_EXIT_SUCCESS,
  CHANNEL_EXIT_FAIL,
  USER_MESSAGE_PRESS,
  USER_MESSAGE_SELECTION_CLEAR,
  MESSAGE_RECEIVED,
  MESSAGE_UPDATED,
  MESSAGE_DELETED,
  CHANNEL_CHANGED,
  TYPING_STATUS_UPDATED,
  READ_RECEIPT_UPDATED,
  OWN_MESSAGE_DELETED,
  OWN_MESSAGE_DELETED_FAIL,
  OWN_MESSAGE_UPDATED,
  OWN_MESSAGE_UPDATED_FAIL,
  MESSAGE_COPY
} from '../actions/types';

const INITAL_STATE = {
  list: [],
  memberCount: 0,
  title: '',
  exit: false,
  typing: '',
  selectedMessages: []
};

const uniqueList = list => {
  return list.reduce((uniqList, currentValue) => {
    let ids = uniqList.map(item => {
      return item.messageId;
    });
    if (ids.indexOf(currentValue.messageId) < 0) {
      uniqList.push(currentValue);
    }
    return uniqList;
  }, []);
};

export default (state = INITAL_STATE, action) => {
  switch (action.type) {
    case INIT_CHAT_SCREEN:
      return { ...state, ...INITAL_STATE };
    case CREATE_CHAT_HANDLER_SUCCESS:
      return { ...state };
    case CREATE_CHAT_HANDLER_FAIL:
      return { ...state };
    case CHANNEL_TITLE_CHANGED:
      return { ...state, title: action.title, memberCount: action.memberCount };
    case CHANNEL_TITLE_CHANGED_FAIL:
      return { ...state };
    case MESSAGE_LIST_SUCCESS:
      return { ...state, list: uniqueList([...state.list, ...action.list]) };
    case MESSAGE_LIST_FAIL:
      return { ...state };
    case SEND_MESSAGE_TEMPORARY:
      return { ...state, list: [...[action.message], ...state.list] };
    case SEND_MESSAGE_SUCCESS:
      const newMessage = action.message;
      let foundNewMessage = false;
      const sendSuccessList = state.list.map(message => {
        if (message.reqId && newMessage.reqId && message.reqId.toString() === newMessage.reqId.toString()) {
          foundNewMessage = true;
          return newMessage;
        } else {
          return message;
        }
      });
      if (foundNewMessage) {
        return { ...state, list: sendSuccessList };
      } else {
        return { ...state, list: [...[newMessage], ...sendSuccessList] };
      }
    case SEND_MESSAGE_FAIL:
      const newChatList = state.list.slice(1);
      return { ...state, list: newChatList };
    case CHANNEL_EXIT_SUCCESS:
      return { ...state, exit: true };
    case CHANNEL_EXIT_FAIL:
      return { ...state, exit: false };
    case USER_MESSAGE_PRESS:
      const newSelectedMessage = action.message;
      return { ...state, selectedMessages: [newSelectedMessage] };
    case USER_MESSAGE_SELECTION_CLEAR:
      return { ...state, selectedMessages: [] };
    case MESSAGE_RECEIVED:
      return { ...state, list: uniqueList([...[action.payload], ...state.list]) };
    case MESSAGE_UPDATED:
      const updatedMessage = action.payload;
      const updatedList = state.list.map(message => {
        if (message.messageId === updatedMessage.messageId) {
          message = updatedMessage;
        }
        return message;
      });
      return { ...state, list: updatedList };
    case MESSAGE_DELETED:
      const deletedList = state.list.filter(message => {
        return message.messageId.toString() !== action.payload.toString();
      });
      return { ...state, list: deletedList };
    case CHANNEL_CHANGED:
      return { ...state, memberCount: action.memberCount, title: action.title };
    case TYPING_STATUS_UPDATED:
      return { ...state, typing: action.typing };
    case READ_RECEIPT_UPDATED:
      return { ...state, list: state.list };
    case OWN_MESSAGE_DELETED_FAIL:
      return { ...state };
    case OWN_MESSAGE_DELETED:
      return { ...state, selectedMessages: [] };
    case OWN_MESSAGE_UPDATED:
      const editedMessage = action.edited;
      const updatedList2 = state.list.map(message => {
        if (message.messageId === editedMessage.messageId) {
          message.isEdited = true;
          message.message = action.contents;
        }
        return message;
      });
      return { ...state, selectedMessages: [], list: updatedList2 };
    case OWN_MESSAGE_UPDATED_FAIL:
      return { ...state };
    case MESSAGE_COPY:
      return { ...state, selectedMessages: [] };
    default:
      return state;
  }
};
