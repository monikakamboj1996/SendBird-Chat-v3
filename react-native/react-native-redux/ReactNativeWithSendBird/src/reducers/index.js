import { combineReducers } from 'redux';
import login from './loginReducer';
import menu from './menuReducer';
import profile from './profileReducer';
import openChannel from './openChannelReducer';
import openChannelCreate from './openChannelCreateReducer';
import chat from './chatReducer';
import member from './memberReducer';
import blockUser from './blockUserReducer';
import groupChannel from './groupChannelReducer';
import groupChannelInvite from './groupChannelInviteReducer';

export default combineReducers({
  login,
  menu,
  profile,
  openChannel,
  openChannelCreate,
  chat,
  member,
  blockUser,
  groupChannel,
  groupChannelInvite
});

export loginReducer from './loginReducer';
export menuReducer from './menuReducer';
export profileReducer from './profileReducer';
export openChannelReducer from './openChannelReducer';
export openChannelCreateReducer from './openChannelCreateReducer';
export chatReducer from './chatReducer';
export memberReducer from './memberReducer';
export blockUserReducer from './blockUserReducer';
export groupChannelReducer from './groupChannelReducer';
export groupChannelInviteReducer from './groupChannelInviteReducer';
