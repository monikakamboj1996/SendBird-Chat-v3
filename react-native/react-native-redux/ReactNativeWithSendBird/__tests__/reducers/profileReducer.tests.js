import { profileReducer } from '../../src/reducers';
import {
  INIT_PROFILE,
  GET_PROFILE_SUCCESS,
  UPDATE_PROFILE_SUCCESS,
  UPDATE_PROFILE_FAIL
} from '../../src/actions/types';

describe('reducers/profileReducer', () => {
  test('checking initial state', () => {
    const action = { type: 'nonexistent' };
    expect(profileReducer(undefined, action)).toMatchSnapshot();
  });

  test('INIT_PROFILE', () => {
    const action = { type: INIT_PROFILE };
    expect(profileReducer(undefined, action)).toMatchSnapshot();
  });

  test('GET_PROFILE_SUCCESS', () => {
    const action = { type: GET_PROFILE_SUCCESS, userInfo: 'userInfo' };
    expect(profileReducer(undefined, action)).toMatchSnapshot();
  });

  test('UPDATE_PROFILE_SUCCESS', () => {
    const action = { type: UPDATE_PROFILE_SUCCESS };
    expect(profileReducer(undefined, action)).toMatchSnapshot();
  });

  test('UPDATE_PROFILE_FAIL', () => {
    const action = { type: UPDATE_PROFILE_SUCCESS, error: 'error' };
    expect(profileReducer(undefined, action)).toMatchSnapshot();
  });
});
