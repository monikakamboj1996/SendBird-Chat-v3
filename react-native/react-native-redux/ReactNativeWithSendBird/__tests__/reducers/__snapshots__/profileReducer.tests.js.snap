// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/profileReducer GET_PROFILE_SUCCESS 1`] = `
Object {
  "error": "",
  "isSaved": false,
  "userInfo": "userInfo",
}
`;

exports[`reducers/profileReducer INIT_PROFILE 1`] = `
Object {
  "error": "",
  "isSaved": false,
  "userInfo": null,
}
`;

exports[`reducers/profileReducer UPDATE_PROFILE_FAIL 1`] = `
Object {
  "error": "",
  "isSaved": true,
  "userInfo": null,
}
`;

exports[`reducers/profileReducer UPDATE_PROFILE_SUCCESS 1`] = `
Object {
  "error": "",
  "isSaved": true,
  "userInfo": null,
}
`;

exports[`reducers/profileReducer checking initial state 1`] = `
Object {
  "error": "",
  "isSaved": false,
  "userInfo": null,
}
`;
